pywren/future.py

```diff
@@ -103,10 +100,10 @@ def result(self, timeout=None, check_only=False, throw_except=True):
             else:
                 return None
 
+        if storage_handler is None:
```

Comment:
Yes, we (Eric and I) agreed that we want futures to be serializable. Storing storage handler as a field of future will make it un-serializable, my previous solution was to have a `save()` method in which we first set `future.storage_handler=None` to make future serializable. We then agree that having `pickle.dump(futures)` to work is also important.